src/helpers/figures/charts/gauge_chart_rendering.ts

```diff
@@ -0,0 +1,466 @@
+  }
+
+  return inflectionValues;
+}
+
+function getGaugeColor(runtime: GaugeChartRuntime): Color {
+  const gaugeValue = runtime.gaugeValue?.value;
+  if (gaugeValue === undefined) {
+    return GAUGE_BACKGROUND_COLOR;
+  }
+  let colorIndex = 0;
+  while (runtime.inflectionValues[colorIndex]?.value <= gaugeValue) {
+    colorIndex++;
+  }
+  return runtime.colors[colorIndex];
+}
+
+function degreesToRadians(degrees: number) {
+  return (degrees * Math.PI) / 180;
+}
+
+function getContrastedTextColor(backgroundColor: Color) {
+  return relativeLuminance(backgroundColor) > 0.3
+    ? GAUGE_TEXT_COLOR
+    : GAUGE_TEXT_COLOR_HIGH_CONTRAST;
+}
+
+function getSegmentsOfRectangle(rectangle: UnalignedRectangle): Segment[] {
+  return [
+    { start: rectangle.topLeft, end: rectangle.topRight },
+    { start: rectangle.topRight, end: rectangle.bottomRight },
+    { start: rectangle.bottomRight, end: rectangle.bottomLeft },
+    { start: rectangle.bottomLeft, end: rectangle.topLeft },
+  ];
+}
+
+function doSegmentIntersect(segment1: Segment, segment2: Segment): boolean {
+  const A = segment1.start;
+  const B = segment1.end;
+  const C = segment2.start;
+  const D = segment2.end;
+
+  const angleSegment1 = Math.atan2(B.y - A.y, B.x - A.x);
+  const angleSegment2 = Math.atan2(D.y - C.y, D.x - C.x);
+  const angleDiff = Math.abs(angleSegment1 - angleSegment2);
+
+  if (angleDiff === 0) {
+    return doColinearSegmentsIntersect(segment1, segment2);
+  }
+
+  /**
+   * Line segment intersection algorithm
+   * https://bryceboe.com/2006/10/23/line-segment-intersection-algorithm/
+   */
+  function ccw(a: PixelPosition, b: PixelPosition, c: PixelPosition) {
+    return (c.y - a.y) * (b.x - a.x) > (b.y - a.y) * (c.x - a.x);
+  }
+
+  return ccw(A, C, D) !== ccw(B, C, D) && ccw(A, B, C) !== ccw(A, B, D);
+}
+
+function doColinearSegmentsIntersect(segment1: Segment, segment2: Segment): boolean {
+  /** check if point b is on segment ac. a, b, c should be colinear*/
+  function onSegment(a: PixelPosition, b: PixelPosition, c: PixelPosition) {
+    return (
+      b.x <= Math.max(a.x, c.x) &&
+      b.x >= Math.min(a.x, c.x) &&
+      b.y <= Math.max(a.y, c.y) &&
+      b.y >= Math.min(a.y, c.y)
+    );
+  }
+
+  return (
+    onSegment(segment1.start, segment1.end, segment2.start) ||
+    onSegment(segment1.start, segment1.end, segment2.end)
+  );
+}
+
+function doRectanglesIntersect(rect1: UnalignedRectangle, rect2: UnalignedRectangle): boolean {
+  const segments1 = getSegmentsOfRectangle(rect1);
+  const segments2 = getSegmentsOfRectangle(rect2);
+  for (const segment1 of segments1) {
+    for (const segment2 of segments2) {
+      if (doSegmentIntersect(segment1, segment2)) {
+        return true;
+      }
+    }
+  }
+  return false;
+}
+
+/**
+ *  Get the rectangle that is tangent to a circle at a given angle.
+ *
+ * @param angle angle between X axis and the point where the rectangle is tangent to the circle
+ */
+function getRectangleTangentToCircle(
```

Comment:
Nitpick : you could have computed and stored `Math.cos(angle)` (ie as `cos`) and `Math.sin(angle)` (as `sin`) and then do something like :
```js
const x = cos * radius;
const y = sin * radius;
const x2 = sin * rectWidth / 2; //sin and cos are inverted as we use complementary angle
const y2 = cos * rectWidth / 2;
const xp = cos * (radius + rectHeight);
const yp = sin * (radius + rectHeight);
```